tidings/utils.py

```diff
@@ -58,17 +20,32 @@ def collate(*iterables, **kwargs):
     descending order rather than ascending.
 
     """
-    # TODO: Liberate into the stdlib.
     key = kwargs.pop('key', lambda a: a)
     reverse = kwargs.pop('reverse', False)
-
     min_or_max = max if reverse else min
-    peekables = [peekable(it) for it in iterables]
```

Comment:
The original seems a lot more readable to me. `peekable`, meanwhile, has indeed been liberated into the `more-itertools` library, which is well-maintained. In fact, [`collate` has been uplifted as well](https://more-itertools.readthedocs.io/en/latest/api.html#more_itertools.collate). It seems like all we really need to do is to wrap the `p` in `(key(p.peek()), p)` in a little jigger that makes it sortable, by reporting that every instance is equal. What I would campaign for is writing one of those wrappers, adding it to more-itertools itself (for which I can expedite a review and release), and depending on more-itertools here for collate and peekable. What do you think?